src/components/PopupWithConfirmation.js

```diff
@@ -0,0 +1,31 @@
+import Popup from "./Popup.js";
+
+export default class PopupWithConfirmation extends Popup {
+  constructor(popupSelector, handleDeleteConfirmSubmit) {
+    super({ popupSelector });
+    this._popupForm = this._popupElement.querySelector(
+      "#confirm-image-delete-popup-form"
+    );
+    this._handleDeleteConfirmSubmit = handleDeleteConfirmSubmit;
+  }
+
+  // this function accepts an function as an argument, and assigns it to this
+  setSubmitAction(action) {
+    this._handleSubmitCallback = action;
+  }
+
+  setEventListeners() {
+    // set event listener with this._handleSubmitCallback as it's handler
+    // don't forget to call super.setEventListeners
+    this._popupForm.addEventListener("submit", (evt) => {
+      evt.preventDefault();
+      this._handleSubmitCallback();
+    });
+    super.setEventListeners();
+  }
+
+  setLoading(isLoading, buttonValue) {
+    this._submitButton.textContent = isLoading ? buttonValue : this._submitButtonValue;
```

Comment:
Needs correction: there is an error in the console when I try to remove a card `PopupWithConfirmation.js:28 Uncaught TypeError: Cannot set properties of undefined (setting 'textContent')
    at PopupWithConfirmation.setLoading (PopupWithConfirmation.js:28:35)
    at PopupWithConfirmation._handleSubmitCallback (index.js:193:24)
    at HTMLFormElement.<anonymous> (PopupWithConfirmation.js:22:12)`